nx/lib/nx/lin_alg.ex

```diff
@@ -258,14 +257,20 @@ defmodule Nx.LinAlg do
     # The idea is that by dividing the tensor by it, large values of
     # tensor entries and large values of p are reduced, which in turn
     # avoids numerical overflow.
-    numerical_stability_coefficient = Nx.reduce_max(t)
-
-    abs_t
-    |> Nx.divide(numerical_stability_coefficient)
-    |> Nx.power(ord)
-    |> Nx.sum(opts)
-    |> Nx.power(inv_ord)
-    |> Nx.multiply(numerical_stability_coefficient)
+    numerical_stability_coefficient = Nx.reduce_max(abs_t)
+
+    if numerical_stability_coefficient == Nx.tensor(0, type: type) do
```

Comment:
Unfortunately this won't work when norm is called from a `defn`. In this case, `Nx.reduce_max` will return an expression, and that will never be equal to 0, and then we will still get an error. The good news is that this will be fixed when we actually support unifity, b

Reply:
Right, right. I had a feeling that something was off with this. I reverted the changes, apart from the abs value issue when calculating `numerical_stability_coefficient` which will fix at least some cases.